monkey/infection_monkey/exploit/hadoop.py

```diff
@@ -90,7 +126,15 @@ def exploit(self, url, command):
         resp = requests.post(
             posixpath.join(url, "ws/v1/cluster/apps/"), json=payload, timeout=LONG_REQUEST_TIMEOUT
         )
-        return resp.status_code == 202
+
+        success = resp.status_code == 202
+        if success:
+            self._publish_exploitation_event(
+                target=self.host.ip_addr,
+                exploitation_success=True,
+                tags=EXPLOIT_TAGS,
+            )
```

Comment:
Timestamps need to be recorded as close to the actual event as possible.

Reply:
What about on failure?

```suggestion
        self._publish_exploitation_event(
            target=self.host.ip_addr,
            exploitation_success=success,
            tags=EXPLOIT_TAGS,
        )
```